warrior/WarriorCore/exec_type_driver.py

```diff
@@ -11,65 +11,349 @@
 limitations under the License.
 '''
 
-
-"""Module that handles the actions to be performed for 
+"""Module that handles the actions to be performed for
 conditional execution of a step in Testcase/Suite/Project """
 
-import sys
 from Framework.Utils.data_Utils import get_object_from_datarepository
 from Framework.Utils.print_Utils import print_error, print_info
 from Classes.argument_datatype_class import ArgumentDatatype
+from Framework.Utils.testcase_Utils import pNote
 
-def main(exectype_elem_node):
-    exec_node = exectype_elem_node.find('Execute')
-    if exec_node is None or exec_node is False:
-        return True, None
+class ElseException(Exception):
+    """
+        Raise an exception when a rule fails with individual else action
+    """
+    def __init__(self, else_action):
+        """Custom exception that used to break through the rule parsing loop"""
+        super(ElseException, self).__init__(else_action)
+        self.action = else_action
+
+def math_decision(exec_condition, exec_cond_var, operator):
+    """
+        Handle the math operator decision
+        :param:
+            exec_condition: value from data_repo to be compared
+            exec_cond_var: user provided value to be compared
+            operator: math operator in plain English
+        :return:
+            True if operator condition match
+            for repo value on left and user value on right
+            Else False
+    """
+    operator = operator.lower()
+    if operator == "ge":
+        return True if get_object_from_datarepository(exec_condition) >= exec_cond_var else False
+    elif operator == "gt":
+        return True if get_object_from_datarepository(exec_condition) > exec_cond_var else False
+    elif operator == "le":
+        return True if get_object_from_datarepository(exec_condition) <= exec_cond_var else False
+    elif operator == "lt":
+        return True if get_object_from_datarepository(exec_condition) < exec_cond_var else False
+
+    pNote("Unknown error occur when deciding value, please check condition value of the step",
+          "Error")
+    return False
```

Comment:
If there are any changes in the the name of get_object_from_datarepository or the parameter names, we have to make changes in multiple places. Instead the following code would be more efficient in my opinion:
do_operator = {'ge': lambda x,y: x>=y,
                          'gt': lambda x,y: x>y,
                          'le': lambda x,y: x<=y,
                          'lt': lambda x,y: x<y,
                          }
return do_operator[operator](get_object_from_datarepository(exec_condition), exec_cond_var)

Reply:
verify_data in data_Utils.py also uses similar logic which can be shared as a common function and re-used instead duplicating in multiple places.

Reply:
I will put this in #245

Reply:
Change has been made in
https://github.com/warriorframework/warriorframework/pull/245